pkg/relayer/config/types.go

```diff
@@ -0,0 +1,132 @@
+	Url            string                                      `yaml:"url"`
+	Authentication YAMLRelayMinerSupplierServiceAuthentication `yaml:"authentication,omitempty"`
+	Headers        map[string]string                           `yaml:"headers,omitempty"`
+}
+
+// YAMLRelayMinerSupplierServiceAuthentication is the structure used to unmarshal
+// the supplier service basic auth of the RelayMiner config file when the
+// supplier is of type "http"
+type YAMLRelayMinerSupplierServiceAuthentication struct {
+	Username string `yaml:"username,omitempty"`
+	Password string `yaml:"password,omitempty"`
+}
+
+// RelayMinerConfig is the structure describing the RelayMiner config
+type RelayMinerConfig struct {
+	SigningKeyName string
+	SmtStorePath   string
+	Pocket         *RelayMinerPocketConfig
+	Proxies        map[string]*RelayMinerProxyConfig
+}
+
+// RelayMinerPocketConfig is the structure resulting from parsing the pocket
+// node URLs section of the RelayMiner config file
+type RelayMinerPocketConfig struct {
+	QueryNodeGRPCUrl *url.URL
+	TxNodeGRPCUrl    *url.URL
+	QueryNodeRPCUrl  *url.URL
+}
+
+// RelayMinerProxyConfig is the structure resulting from parsing the proxy
+// section of the RelayMiner config file.
+// Each proxy embeds a map of supplier configs that are associated with it.
+// Other proxy types may embed other fields in the future. eg. "https" may
+// embed a TLS config.
+type RelayMinerProxyConfig struct {
+	// Name is the name of the proxy server, used to identify it in the config
+	Name string
+	// Host is the host on which the proxy server will listen for incoming
+	// relay requests
+	Host string
+	// Type is the transport protocol used by the proxy server like (http, https, etc.)
+	Type string
+	// Suppliers is a map of serviceIds -> RelayMinerSupplierConfig
+	Suppliers map[string]*RelayMinerSupplierConfig
+	// XForwardedHostLookup is a flag that indicates whether the proxy server
+	// should lookup the host from the X-Forwarded-Host header before falling
+	// back to the Host header.
+	XForwardedHostLookup bool
+}
+
+// RelayMinerSupplierConfig is the structure resulting from parsing the supplier
+// section of the RelayMiner config file.
+type RelayMinerSupplierConfig struct {
+	// Name is the serviceId corresponding to the current configuration.
+	Name string
+	// Type is the transport protocol used by the supplier, it must match the
+	// type of the proxy it is associated with.
+	Type string
+	// Hosts is a list of hosts advertised on-chain by the supplier, the corresponding
+	// proxy server will accept relay requests for these hosts.
+	Hosts []string
+	// ServiceConfig is the config of the service that relays will be proxied to.
+	ServiceConfig *RelayMinerSupplierServiceConfig
+}
+
+// RelayMinerSupplierServiceConfig is the structure resulting from parsing the supplier
+// service sub-section of the RelayMiner config file.
+// If the supplier is of type "http", it may embed a basic auth structure and
```

Comment:
I think this is TMI (Too much Information). The reason we have attribute specific comments is to capture this sort of detail.